### PR TITLE
Fix VecExtractDictObs does not handle terminal observation

### LINKED ISSUE #1441: [Bug]: VecExtractDictObs does not handle terminal observation
### 🐛 Bug

VecExtractDictObs Wrapper currently does not update the ``terminal_observation`` entry in the info returned by VecEnv.
This causes an exception when storing into a buffer, since the observation shape mismatches. 

### To Reproduce

```python
from stable_baselines3 import DQN
from stable_baselines3.common.vec_env import VecEnvWrapper, DummyVecEnv, VecExtractDictObs, VecEnv
from gym import spaces
from gym import make

class VecObsToDict(VecEnvWrapper):
    def __init__(self, venv: VecEnv):
        super().__init__(venv, spaces.Dict({"observation": venv.observation_space}))

    def step_wait(self):
        obs, rewards, dones, infos = self.venv.step_wait()
        obs = {"observation": obs}
        for info in infos:
            if "terminal_observation" in info:
                info["terminal_observation"] = {"observation": info["terminal_observation"]}
        return obs, rewards, dones, infos
    
    def reset(self):
        obs = self.venv.reset()
        obs = {"observation": obs}
        return obs

env = make("CartPole-v1")
venv = DummyVecEnv([lambda: env])
venv = VecObsToDict(venv)
venv = VecExtractDictObs(venv, "observation")

dqn = DQN("MlpPolicy", venv)
dqn.learn(100000, progress_bar=True)
```

### Relevant log output / Error message

```shell
has occurred: TypeError       (note: full exception trace is shown but execution is paused at: _run_module_as_main)
float() argument must be a string or a real number, not 'dict'
  File "C:\Users\samue\miniconda3\envs\thesis\lib\site-packages\stable_baselines3\common\off_policy_algorithm.py", line 466, in _store_transition
    next_obs[i] = infos[i]["terminal_observation"]
  File "C:\Users\samue\miniconda3\envs\thesis\lib\site-packages\stable_baselines3\common\off_policy_algorithm.py", line 558, in collect_rollouts
    self._store_transition(replay_buffer, buffer_actions, new_obs, rewards, dones, infos)
  File "C:\Users\samue\miniconda3\envs\thesis\lib\site-packages\stable_baselines3\common\off_policy_algorithm.py", line 311, in learn
    rollout = self.collect_rollouts(
  File "C:\Users\samue\miniconda3\envs\thesis\lib\site-packages\stable_baselines3\dqn\dqn.py", line 269, in learn
    return super().learn(
  File "C:\Users\samue\Documents\GitHub\thesis\bug.py", line 29, in <module>
    dqn.learn(100000, progress_bar=True)
  File "C:\Users\samue\miniconda3\envs\thesis\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\samue\miniconda3\envs\thesis\lib\runpy.py", line 196, in _run_module_as_main (Current frame)
    return _run_code(code, main_globals, None,
TypeError: float() argument must be a string or a real number, not 'dict'
```

### System Info

- OS: Windows-10-10.0.22621-SP0 10.0.22621
- Python: 3.10.9
- Stable-Baselines3: 1.8.0
- PyTorch: 2.0.0
- GPU Enabled: True
- Numpy: 1.23.5
- Gym: 0.21.0

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a minimal working example to reproduce the bug
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

### PR DESCRIPTION
**Pull request by WeberSamuel**
This pull request add handling of terminal observation in ``VecExtractDictObs``

## Motivation and Context
closes #1441
- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)
- [ ] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: docs/misc/changelog.rst [hunk 1]
```diff
@@ -64,6 +63,7 @@ Bug Fixes:
 - Fixed PPO train/n_updates metric not accounting for early stopping (@adamfrly)
 - Fixed loading of normalized image-based environments
 - Fixed ``DictRolloutBuffer.add`` with multidimensional action space (@younik)
+- Fixed ``VecExtractDictObs`` does not handle terminal observation (@WeberSamuel)
```
Comment (WeberSamuel) [inline docs/misc/changelog.rst:66]: I wasn't sure if this is the right place. Because this version is already released

Comment (araffin) [reply docs/misc/changelog.rst:66]: no sorry, if you re-open the PR with another branch I could fix that.

Comment (araffin) [reply docs/misc/changelog.rst:66]: (master is protected by default)

Comment (WeberSamuel) [reply docs/misc/changelog.rst:66]: okay, I will reopen the PR

**Pull request closed without merging**